src/useCases/createUser.ts

```diff
@@ -91,11 +99,19 @@ export default async (connection: Database, currentUserId: number, userDetails:
         console.error(insertUserResult)
         return Error("Could not insert record into users table")
       }
-
-      if (userDetails.groupId) {
-        const userGroupResult = await insertUserIntoGroup(task, insertUserResult.id, currentUserId, [
-          userDetails.groupId
-        ])
+      const groups = await getUserGroups(connection, [currentUser.username ?? ""])
+      if (isError(groups)) {
+        return groups
+      }
+      const filteredGroups = groups.filter((group) => userDetails[group.name] === "yes")
+      const userDetailsWithGroups = { ...userDetails, groups: filteredGroups }
+      if (userDetailsWithGroups.groups && userDetailsWithGroups.groups.length > 0) {
+        const userGroupResult = await insertUserIntoGroup(
+          task,
+          insertUserResult.id,
+          currentUser.id ?? -1,
```

Comment:
can we return an error earlier if we do not have a currentUser.id? i.e. not have the SQL query return a failure in this case and handle this more elegantly?

Reply:
Review comment 6